File: blog/go-design/Person.java
public class Person {
  public String name;
  public void say(String input) {
    System.out.println(this.name + " says: " + input);
  }

  public static void main(String[] args) {
    Person p = new Person();
    p.name = "Alex";
    p.say("Hello World!");
  }
}

File: blog/go-design/person.c
#include "stdio.h"
#include "string.h"

struct person {
  char* name;
}

void say(struct person p, char* s) {
  printf("%s says: %s\n", p.name, s);
}

int main(int argc, char const *argv[]) {
  struct person Alex;
  strcpy(Alex.name, "Alex");

  say(Alex, "Hello World!");
  return 0;
}
